airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/JobNotifier.java

```diff
@@ -103,7 +104,14 @@ public void failJob(final String reason, final Job job) {
         try {
           final Builder<String, Object> notificationMetadata = ImmutableMap.builder();
           notificationMetadata.put("connection_id", connectionId);
-          notificationMetadata.put("notification_type", notification.getNotificationType());
+          if (notification.getNotificationType().equals(NotificationType.SLACK) &&
```

Comment:
This tracking code pollutes the logic of the function. can we put it in a separate sub-function?